bootstrap

```diff
@@ -45,12 +45,17 @@ Linux)
         fi
         ;;
     RedHatEnterpriseWorkstation|RedHatEnterpriseServer|CentOS)
-        for package in python2-pip python-virtualenv mariadb-devel libev-devel libvirt-devel libffi-devel; do
-	    if [ "$(rpm -q $package)" == "package $package is not installed" ]; then
-		missing="${missing:+$missing }$package"
-	    fi
+        if [[ "$PYTHON" =~ "python2" ]]; then
+            packages=(python-pip python-devel python-virtualenv mariadb-devel libev-devel libvirt-devel libffi-devel)
```

Comment:
what's the advantage of using array here?

Reply:
I don't this is an array here. I think this is a conditional construct. [ref here](http://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Conditional-Constructs).

Reply:
i don't think i am talking about `if`. what i am referencing is the above line.
```sh
packages=(....)
```

> Arrays are assigned to using compound assignments of the form
> name=(value1 value2 … )

see https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Arrays

Reply:
ok. correct me If I am wrong. the pattern I followed is similar to that one where the comparison for previous distros already exists. [some thing like this](https://github.com/ceph/teuthology/blob/master/bootstrap#L24)

Reply:
> what's the advantage of using array here?

Why not using array for lists? It is natural and allows to generalize iteration algorithms:
```
a=("abc" "a b c")

for i in "${a[@]}" ; do
  echo - $i
done

b='xyz "x y z"'
for i in $b ; do
  echo + $i
done
```
will give:
```
- abc
- a b c
+ xyz
+ "x
+ y
+ z"
```

Also, each element of array can be taken by index.

Reply:
> ok. correct me If I am wrong. the pattern I followed is similar to that one where the comparison for previous distros already exists. [some thing like this](https://github.com/ceph/teuthology/blob/master/bootstrap#L24)

Maybe use the same pattern as you refer to, like use `$deps` instead `$packages`?

Reply:
done. changed `deps`

Reply:
Resolving, I think rakesh is following same patter what we have in the rest of the places in bootstrap.